src/write.jl

```diff
@@ -2,28 +2,52 @@
 ##### utilities
 #####
 
-_edf_write(io::IO, value::Union{String,Char}) = Base.write(io, value)
-_edf_write(io::IO, date::Date) = Base.write(io, uppercase(Dates.format(date, dateformat"dd-u-yyyy")))
-_edf_write(io::IO, date::DateTime) = Base.write(io, Dates.format(date, dateformat"dd\.mm\.yyHH\.MM\.SS"))
-_edf_write(io::IO, x::Integer) = Base.write(io, string(x))
-_edf_write(io::IO, x::AbstractFloat) = Base.write(io, string(isinteger(x) ? trunc(Int, x) : x))
+_edf_repr(value::Union{String,Char}) = value
+_edf_repr(date::Date) = uppercase(Dates.format(date, dateformat"dd-u-yyyy"))
+_edf_repr(date::DateTime) = Dates.format(date, dateformat"dd\.mm\.yyHH\.MM\.SS")
 
-function _edf_write(io::IO, metadata::T) where T<:Union{PatientID,RecordingID}
-    bytes_written = 0
-    for name in fieldnames(T)
-        field = getfield(metadata, name)
-        if T <: RecordingID && name === :startdate
-            bytes_written += _edf_write(io, "Startdate ")
+function _edf_repr(x::Real)
+    if 0 < abs(x) <= 6e-8
+        error("EDF.jl does not yet support writing numbers where `0 < abs(x) <= 6e-8`: $x")
+    end
+    if isinteger(x)
+        str = string(trunc(Int, x))
+        length(str) <= 8 && return str
+    else
+        fpart, ipart = modf(x)
+        ipart_str = string(Int(ipart))
+        fpart_str = @sprintf "%.7f" fpart
+        if (fpart < 0 || ipart < 0)
+            if !startswith(ipart_str, '-')  # add leading `-`, if missing
```

Comment:
You can combine this and the next condition, since if one is negative then they both are